packages/webcomponents/src/components/business-forms/payment-provisioning/payment-provisioning.tsx

```diff
@@ -17,51 +21,112 @@ import { ComponentError, ComponentErrorCodes, ComponentErrorSeverity } from '../
 })
 export class PaymentProvisioning {
   @State() formLoading: boolean = false;
+  @State() businessProvisioned: boolean = false;
   @State() currentStep: number = 0;
   @State() errorMessage: string;
-  
-  @Prop() authToken: string;
+  @State() postProvisioning: Function;
+  @State() getBusiness: Function;
+
   @Prop() businessId: string;
+  @Prop() authToken: string;
   @Prop() allowOptionalFields?: boolean = false;
   @Prop() formTitle?: string = 'Business Information';
   @Prop() removeTitle?: boolean = false;
- 
+
   @Event({ eventName: 'click-event' }) clickEvent: EventEmitter<BusinessFormClickEvent>;
   @Event({ eventName: 'error-event' }) errorEvent: EventEmitter<ComponentError>;
+  @Event() submitted: EventEmitter<BusinessFormSubmitEvent>;
 
   analytics: JustifiAnalytics;
 
   componentWillLoad() {
     this.analytics = new JustifiAnalytics(this);
-    if (!this.businessId || !this.authToken) {
-      this.errorMessage = 'Invalid business id or auth token';
+    this.initializeApi();
+    this.fetchBusiness();
+
+    this.refs = [this.coreInfoRef, this.legalAddressRef, this.additionalQuestionsRef, this.representativeRef, this.ownersRef, this.bankAccountRef, this.documentUploadRef, this.termsRef];
+  }
+
+  disconnectedCallback() {
+    this.analytics.cleanup();
+  }
+
+  private initializeApi() {
+    if (this.authToken && this.businessId) {
+      this.getBusiness = makeGetBusiness({
+        authToken: this.authToken,
+        businessId: this.businessId,
+        service: new BusinessService()
+      });
+      this.postProvisioning = makePostProvisioning({
+        authToken: this.authToken,
+        businessId: this.businessId,
+        product: 'payment',
+        service: new ProvisionService()
+      });
+    } else {
       this.errorEvent.emit({
+        message: 'auth-token and business-id are required',
         errorCode: ComponentErrorCodes.MISSING_PROPS,
-        message: this.errorMessage,
         severity: ComponentErrorSeverity.ERROR,
       });
-      return;
     }
-
-    this.refs = [this.coreInfoRef, this.legalAddressRef, this.additionalQuestionsRef, this.representativeRef, this.ownersRef, this.bankAccountRef, this.documentUploadRef, this.termsRef];
   }
 
-  disconnectedCallback() {
-    this.analytics.cleanup();
+  fetchBusiness() {
```

Comment:
Maybe I would rename this function to something like `setBusinessProvisioned` which feels more accurate with what the function is doing.